shared-libs/transitions/src/schedule/due_tasks.js

```diff
@@ -35,15 +35,15 @@ const getTemplateContext = (doc, callback) => {
 
 module.exports = {
   execute: callback => {
-    var now = moment(date.getDate()),
-      overdue = now.clone().subtract(7, 'days');
+    const now = moment(date.getDate());
+    const overdue = now.clone().subtract(7, 'days');
 
     db.medic.query(
-      'medic/due_tasks',
+      'medic/messages_by_state',
```

Comment:
The functionality changes slightly here, maybe for the better?
The old view emitted `doc.scheduled_tasks[x].due`, so scheduled tasks ONLY and didn't "calculate" due, used the value directly.
The new view emits tasks and scheduled_tasks and the value is calculated as "task.due || task.timestamp || doc.reported_date".

Reply:
> scheduled tasks ONLY

I don't think it's possible for `tasks` to be in the "scheduled" state, is it? That was my assumption.

> didn't "calculate" due, used the value directly.

I think in practice all scheduled tasks have a `due` date so it should be equivalent.

This was just an attempt to make the views more reusable. I really wanted to move this one into the `medic-sms` ddoc but because it's called by the schedule all the time whether or not you use SMS it would force the reindexing of all the medic-sms views, so I left it where it was. It'd be helpful if you could turn schedules on/off...

Reply:
> I don't think it's possible for tasks to be in the "scheduled" state, is it? 

This is a very reasonable assumption. 
I found another issue, though. 
The `messages_by_state` view only indexes already existent messages (that are already generated), while a task that is scheduled might not even have a `messages` property: 
```
{
      "due": "2019-06-12T09:00:00.000Z",
      "group": 1,
      "type": "ANC Reminders LMP",
      "translation_key": "schedule.anc_lmp",
      "message_key": "messages.schedule.anc.reminder",
      "recipient": "clinic",
      "state_history": [
        {
          "state": "scheduled",
          "timestamp": "2019-06-14T06:03:17.365Z"
        }
      ],
      "state": "scheduled"
    },
```

Reply:
Absolutely right. I've added a switch to emit when there are no messages too.

Reply:
Sorry! I'll try again...